src/wesnothd_connection.cpp

```diff
@@ -564,3 +569,24 @@ bool wesnothd_connection::wait_and_receive_data(config& data)
 
 	return receive_data(data);
 };
+
+void wesnothd_connection::set_keepalive(int seconds)
+{
+	boost::asio::socket_base::keep_alive option(true);
+	utils::get<raw_socket>(socket_)->set_option(option);
+
+#ifdef __linux__
+	int cnt = 5;
+	int interval = 1;
```

Comment:
What would be a good formula then? I don't want additional preferences for both of these.

Reply:
Added a formula - for every additional 10 seconds, it adds one to the keepalive count instead.

Reply:
Oh, missed your previous message for some reason. I suppose that works as well. Even bigger overshoot for small timeouts though. I think it makes more sense to use the biggest part for the timeout and rather have a small fixed count than interval.

Reply:
The goal was to have it be as close as possible based on the formula loonycyborg gave earlier for what the actual timeout duration would be (`timeout + count*interval`), with the assumption that I can't set `interval` or `count` to 0. With
```
int timeout = 10;
int cnt = std::max((seconds - 10) / 10, 1);
int interval = 10;
```
it means it will match what's in the preferences except for the default value of 10 seconds which will actually result in a timeout of 20 seconds.